code/lib/csf-tools/src/enrichCsf.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-underscore-dangle */
+import * as t from '@babel/types';
+import generate from '@babel/generator';
+import template from '@babel/template';
+import { CsfFile, formatCsf, loadCsf } from './CsfFile';
+
+export const enrichCsf = (csf: CsfFile) => {
+  Object.keys(csf._storyExports).forEach((key) => {
+    const storyExport = csf.getStoryExport(key);
+    const source = extractSource(storyExport);
+    const addParameter = template(`
+       %%key%%.parameters = { storySource: { source: %%source%% }, ...%%key%%.parameters };
+    `)({
+      key: t.identifier(key),
+      source: t.stringLiteral(source),
+    }) as t.Statement;
```

Comment:
Without knowing too much about what goes on behind the scenes, on the surface this looks like something that could be handled with a simple template literal instead of requiring babel/template?

Reply:
https://github.com/storybookjs/storybook/pull/19684

Reply:
Yeah I see those changes. I don't know how much faster it is, but the raw AST is definitely less readable that `template`.